### PR TITLE
feat: support backfill_rate_limit for source backfill

### LINKED ISSUE #19297: Support ALTER BACKFILL_RATE_LMIT for SourceBackfill

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
fix #19297

This is a missing feature. We support source_rate_limit for shared source executor, but cannot rate limit source backfill before this PR.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19445?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#19445** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19445?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19445?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#19443** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19443?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/meta/src/controller/streaming_job.rs [hunk 1]
```diff
@@ -1412,7 +1411,7 @@ impl CatalogController {
         fragments.retain_mut(|(_, fragment_type_mask, stream_node)| {
             let mut found = false;
             if (*fragment_type_mask & PbFragmentTypeFlag::StreamScan as i32 != 0)
-                || (*fragment_type_mask & PbFragmentTypeFlag::Source as i32 != 0)
```
Comment (xxchan) [inline src/meta/src/controller/streaming_job.rs:1415]: The previous code looks wrong: Alter backfill rate limit will affect MV on non-shared source

**Comment by xxchan** [comment]
https://github.com/risingwavelabs/risingwave-docs-new/pull/62

**Review by kwannoel — APPROVED** [review]
LGTM, thanks!

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: proto/stream_plan.proto [hunk 2]
```diff
@@ -231,7 +231,7 @@ message SourceBackfillNode {
   catalog.StreamSourceInfo info = 4;
   string source_name = 5;
   map<string, string> with_properties = 6;
-  // Streaming rate limit
+  // Backfill rate limit
   optional uint32 rate_limit = 7;
```
[the remark this answers is not in the document]
Comment (xxchan) [reply proto/stream_plan.proto:235]: It will be breaking change

[the remark this answers is not in the document]
Comment (tabVersion) [reply proto/stream_plan.proto:235]: why? It keeps the same tag number. https://stackoverflow.com/questions/45431685/protocol-buffer-does-changing-field-name-break-the-message

[the remark this answers is not in the document]
Comment (xxchan) [reply proto/stream_plan.proto:235]: IIRC sql meta backend stores with JSON encoding

**Comment by xxchan** [comment]
### Merge activity

* **Nov 20, 3:57 PM GMT+8**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19445).

**Pull request merged**